test-end-to-end/android-maestro/e2e-tests.disabled/004_007_user_enter_assignment_page_before_graceperiod_but_submission_after_grace_Period.yaml

```diff
@@ -0,0 +1,158 @@
+    env:
+      CONTROLSERVER: ${CONTROLSERVER}
+      TESTSERIAL: ${TESTSERIAL}
+      TESTNAME: ${TESTNAME}
+- runFlow: "../common/clear_state_and_get_started.yaml"
+- runFlow: "../common/select_site.yaml"
+- runFlow:
+    file: "../common/user_login.yaml"
+# add new course
+- runFlow:
+    file: "../common/add_course.yaml"
+    env:
+      TESTCOURSENAME: "Test Course Block"
+# add new teacher using flow - add_new_teacher_inside_course ---------
+- runFlow:
+    file: "../common/add_new_teacher_inside_course.yaml"
+    env:
+      TESTCOURSENAME: ${TESTCOURSENAME}
+      TEACHERFIRSTNAME: "Teacher"
+      TEACHERLASTNAME: "1"
+      USERNAME: "t1"
+      GENDER: "Female"
+      PASSWORD: "test"
+# add 6 students to the course through bulk import using .csv file
+- runFlow:
+    file: "../common/add_6_students_by_bulk_import.yaml"
+    env:
+      TESTFILENAME: "Ustad_Students_List.csv"
+      CONTROLSERVER: ${CONTROLSERVER}
+      TESTSERIAL: ${TESTSERIAL}
+# teacher add an assignment block
+- runFlow: "../common/clear_state_and_get_started.yaml"
+- runFlow: "../common/select_site.yaml"
+- runFlow:
+    file: "../common/user_login.yaml"
+    env:
+      USERNAME: "t1"
+      PASSWORD: "test"
+- tapOn: "Courses"
+- tapOn: "Test Course Block"
+# edit the course to add blocks
+- tapOn:
+    id: "floating_action_button"
+# add assignment block ------ 1
+- scrollUntilVisible:
+    element:
+      text: "Add block"
+    direction: DOWN
+- tapOn: "Add block"
+- swipe:
+    from:
+      text: "Drag handle" # From (x, y) center of element
+    direction: UP # To (50% of width, 10% of height)
+- tapOn: "Assignment"
+- tapOn: "Title*"
+- inputText: "Assignment 1"
+- hideKeyboard
+- tapOn: "Description"
+- tapOn:
+    id: "com.toughra.ustadmobile:id/editor"
+- inputText: "test block"
+- tapOn: "Done"
+- assertVisible: "Don't show before"
+- tapOn:
+    text: "Don't show before"
+- runScript:
+    file: "../common/setDate.js"
+- inputText: ${output.yesterdayDate}
+- tapOn: "hh:mm"
+- runScript:
+    file: "../common/setDate.js"
+- inputText: ${output.delayTime}
+- hideKeyboard
+- scrollUntilVisible:
+    element:
+      text: "Deadline"
+    direction: DOWN
+- tapOn: "Deadline"
+- runScript:
+    file: "../common/setDate.js"
+- inputText: ${output.yesterdayDate}
+- tapOn: "hh:mm"
+- runScript:
+    file: "../common/setDate.js"
+- inputText: ${output.currentTime}
+- hideKeyboard
+- tapOn: "End of grace period"
+- runScript:
+    file: "../common/setDate.js"
+- inputText: ${output.currentDate}
+- tapOn: "hh:mm"
+- runScript:
+    file: "../common/setDate.js"
+- inputText: ${output.testTime}
+- hideKeyboard
+- tapOn: "Late submission penalty"
+- inputText: "10"
+- hideKeyboard
+- scrollUntilVisible:
+    element:
+      text: "Submission policy"
+    direction: DOWN
+- assertVisible: "Must submit all at once"
+- assertVisible: "Done"
+- takeScreenshot: "${TESTRESULTSDIR}/screenshots/added_assignment"
+- tapOn:
+    text: "Done"
+    delay: 1000
```

Comment:
We should **never** have a manual delay without an explanation of why its there. Why do we need this manual delay?